### PR TITLE
Conversation read mark

### PR DESCRIPTION
**Pull request by su-chang**
## Checklist

- [ ] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [ ] Tests for the changes have been added
- [ ] CI has been passed. (GitHub actions all turns green)
- [ ] CLA has been signed

## Description

> please describe the changes that you are making, with the related issue number.

### CODE REVIEW

**Comment by su-chang** [comment]
Related issue: #2166

File: src/user/contact.ts [hunk 1]
```diff
@@ -844,6 +844,23 @@ class Contact extends ContactEventEmitter implements Sayable {
...
+   * Mark the conversation as read
+   * @param {boolean} hasRead
+   *
+   * @example
+   * const bot = new Wechaty()
+   * const contact = await bot.Contact.find({name: 'xxx'})
+   * await contact.readMark()
+   */
+  public async readMark (hasRead?: boolean): Promise<void | boolean> {
+    try {
```
Comment (huan) [inline src/user/contact.ts:857]: Please make the semantic to be clearer:

```ts
if (typeof hasRead === 'undefined') {
  return this.wechaty.puppet.conversationReadMark(this.id)
} else {
  await this.wechaty.puppet.conversationReadMark(this.id, hasRead)
}
```

File: src/user/contact.ts [hunk 2]
```diff
@@ -844,6 +844,23 @@ class Contact extends ContactEventEmitter implements Sayable {
...
+  /**
+   * Mark the conversation as read
+   * @param {boolean} hasRead
+   *
+   * @example
+   * const bot = new Wechaty()
+   * const contact = await bot.Contact.find({name: 'xxx'})
+   * await contact.readMark()
+   */
+  public async readMark (hasRead?: boolean): Promise<void | boolean> {
```
Comment (huan) [inline src/user/contact.ts:856]: Please add overload method definations:

```ts
  async readMark (hasRead: boolean): Promise<void>
  async readMark (): Promise<boolean>
```

File: src/user/room.ts [hunk 3]
```diff
@@ -1023,6 +1023,23 @@ class Room extends RoomEventEmitter implements Sayable {
...
+  /**
+   * Mark the conversation as read
+   * @param {boolean} hasRead
+   *
+   * @example
+   * const bot = new Wechaty()
+   * const room = await bot.Room.find({topic: 'xxx'})
+   * await room.readMark()
+   */
+  public async readMark (hasRead?: boolean): Promise<void | boolean> {
```
Comment (huan) [inline src/user/room.ts:1035]: The same with message reviews.

**Review by huan — CHANGES_REQUESTED** [review]

**Review by huan — APPROVED** [review]
Awesome, thank you very much for the enhancements!

**Pull request merged**